x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-BigArrayBlock.java.st

```diff
@@ -106,7 +109,7 @@ public final class $Type$BigArrayBlock extends AbstractArrayBlock implements $Ty
 
         // The following line is correct because positions with multi-values are never null.
         int expandedPositionCount = vector.getPositionCount();
-        long bitSetRamUsedEstimate = BlockRamUsageEstimator.sizeOfBitSet(expandedPositionCount);
+        long bitSetRamUsedEstimate = Math.max(nullsMask.size(), BlockRamUsageEstimator.sizeOfBitSet(expandedPositionCount));
```

Comment:
This is also new; I realized that sometimes the initial `nullsMask` is longer than necessary (even taking into account that `nullsMask.size()` is a multiple of `Long.SIZE`). Since `shiftNullsToExpandedPositions` now just pre-allocates as much memory for the expanded `nullsMask` as for the initial `nullsMask`, this is slightly more correct.